contracts/VM.sol

```diff
@@ -41,43 +43,57 @@ abstract contract VM {
         bytes memory outdata;
 
         uint256 commandsLength = commands.length;
-        for (uint256 i; i < commandsLength; i=_uncheckedIncrement(i)) {
+        for (uint256 i; i < commandsLength; i = _uncheckedIncrement(i)) {
             command = commands[i];
             flags = uint256(uint8(bytes1(command << 32)));
 
             if (flags & FLAG_EXTENDED_COMMAND != 0) {
-                indices = commands[i++];
+                indices = commands[_uncheckedIncrement(i)];
```

Comment:
With this change, it seems `i` is not incremented in the scope of the looping over the commands. Integers are value types. As we are in a context of `FLAG_EXTENDED_COMMAND`, the next command is used as a 32-byte `in` list of indices, hence we need to skip it and we might want to change to `commands[++i]`

Reply:
you're right!